docs/architecture/components.mdx

```diff
@@ -92,57 +94,60 @@ The Node Collector is also responsible for recording some k8s metrics, and colle
-- Monitor changes to the instrumentation configs, and update the agents accordingly at runtime.
+- Inspect the runtime details of the instrumented sources containers (programming language, relevant env variables, runtime versions, etc).
+- Implement the Kubernetes virtual instrumentation devices. In the presence of a supported devices on the pod, odiglet will mount the agent code from fily-system to the pod, and add the relevant environment variables to initiate instrumentation.
+- For eBPF based instrumentation, odiglet will inject the eBPF code into the pod, pick up the telemetry items from the eBPF queues and push it downstream to node collector component.
+- Monitor changes to instrumentations config, and apply to the relevant agents accordingly at runtime.
 
 ### Node Collector
 
-**Role**: The node collector (in the context of instrumentation) is responsible for collecting k8s metrics, and logs from the node filesystem.
+**Role**: Node collector (in the context of instrumentation) is responsible for scraping Kubernetes metrics, and collect logs from the node filesystem.
 
-- When a destination for logs exists, the node collector will use the ["filelog receiver"](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver) to read the relevant container logs from the node filesystem and export them as otel logs to the cluster collector.
-- When a destination for metrics exists, the node collector will use the ["kubeletstats receiver"](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) to scrape k8s related metrics from the kubelet and export them as otel metrics to the cluster collector.
+- When a destination for metrics exists, the node collector will use the OpenTelemetry-Collector ["kubeletstats receiver"](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) to scrape k8s related metrics from the kubelet and export them as otel metrics to the cluster collector.
+- When a destination for logs exists, the node collector will use the OpenTelemetry-Collector ["filelog receiver"](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver) to read the relevant pod logs from the node filesystem and export them as otel logs to the cluster collector.
 
 ## Odigos
 
 ### Odigos UI
 
-**Type**: The Odigos UI is a web application
+**Type**: Odigos UI is a web application, it's backend is a deployment in the cluster, and the frontend runs in the user's browser.
 
-**K8s Resource**: Deployment / `odigos-ui`
+**K8s Resource**: `kubectl get deployments -n odigos-system odigos-ui`
 
-**Role**: The Odigos UI is a web application that allows users to configure the Odigos pipeline.
+**Role**: The Odigos UI is a web application that allows users to configure various aspects of the Odigos installation.
```

Comment:
```suggestion
**Role**: The Odigos UI is a web application that allows users to configure the Odigos installation and pipeline.
```

I think "pipeline" here actually makes sense. "installation" tells me it's more for like system settings

Reply:
I depends what we call pipeline. are sources part of the pipeline? is intrumentation rules?